cmd/zdap/commands/clone.go

```diff
@@ -131,15 +135,73 @@ func CloneResourceCompletion(c *cli.Context) {
 }
 
 func CloneResource(c *cli.Context) error {
-	clone, err := cloneResource(c.Args().Slice())
+	clone, err := cloneResource(c.Args().Slice(), zdap.ClaimArgs{})
 	if err != nil {
 		return err
 	}
 	fmt.Println("Attach to project by running, run:")
-	fmt.Printf("zdap attach @%s %s %s\n", clone.Server, clone.Resource, clone.CreatedAt.Format(utils.TimestampFormat))
+	fmt.Printf("zdap attach --new=false @%s:%d %s %s\n", clone.Server, clone.Port, clone.Resource, clone.CreatedAt.Format(utils.TimestampFormat))
+	return nil
+}
+
+type ClaimResult struct {
+	Server  string `json:"server"`
+	Port    int    `json:"port"`
+	CloneId string `json:"clone_id"`
+}
+
+func ExpireClaimedResource(c *cli.Context) error {
+	args := c.Args().Slice()
+	if len(args) < 2 {
+		return errors.New("no claim specified")
+	}
+
+	resource := args[0]
+	claimId := args[1]
+
+	var err error
+
+	cfg, err := getConfig()
+	if err != nil {
+		return err
+	}
+
+	servers := cfg.Servers
+
+	for _, s := range servers {
+		client := zdap.NewClient(cfg.User, s)
+		err = client.ExpireClaim(resource, claimId)
+		if err != nil {
+			fmt.Println("[Err]", err)
+			err = nil
+			continue
+		}
+	}
+
 	return nil
+
 }
-func cloneResource(args []string) (*zdap.PublicClone, error) {
+
+func ClaimResource(c *cli.Context) error {
+	ttl := c.Int64("ttl")
+	clone, err := cloneResource(c.Args().Slice(), zdap.ClaimArgs{
+		ClaimPooled: true,
+		TtlSeconds:  ttl,
+	})
+	if err != nil {
+		return err
+	}
+	b, err := json.Marshal(ClaimResult{
+		Server:  clone.Server,
+		Port:    clone.Port,
+		CloneId: clone.Name,
```

Comment:
If we include the server name/ip in the `CloneId`, we can use that in the `zdap expire` command to know what server hosts the clone, so we don't need to iterate over all the zdap server instances and try to delete a clone.

Reply:
Sure, but should we really include it in the same field? We could put it in a separate field so that no one has to parse the id to separate the server from the name of the dataset.

Reply:
Sure, we could do it as a separate parameter also, but it is one extra parameter for the consumer of `zdap claim` to parse and store, to be able to pass that to `zdap expire`. Should we support `zdap expire` command with only the clone name as parameter?

Since we call it `CloneId` and not `CloneName`, I think it would be acceptable to add the server ip (like `<clone_name>@<sever_ip>`) when it's written to the console, and `zdap expire` would then separate the passed `CloneId` on `@` to get the server that host the clone, so we don't have to iterate over all server and try to expire/delete a clone that may not exist on that server. We also don't verify that the user calling `zdap expire` actually owns the clone as far as I can see. So in worst case, as it is right now, it is a small risk that another users clone can be unintentionally expired/deleted.

Reply:
Since the id is the name of the dataset I think the risk of unintentionally expiring someone elses clone is pretty negligible. 

I'm thinking the API will be clearer if the CloneId can be used immediately as an id in the expire request. If we want to go the route of CloneId being `<clone_name>@<server_ip>` I think the API should also be able to accept that whole thing as CloneId. Then the client can decide to parse the id and separate it if it wants, but it could also just send that thing to all servers. (the servers could also then verify that the server_ip matches before attempting to expire anything)

Reply:
Yes, I agree, I did say that is were a small risk.

It depends on what API you are talking about, we are somewhat discussing both the `zdap` cli API, and `zdapd` API.
As I see it, this is only related to the `zdap` cli, if we use the zdap (`client.go`) as a dependency, I totally agree that server should be a separate parameter.

And on that note, the `zdapd` API isn't consistent in it's naming of the new pool related API, since it return `Name` when cloning, but call that `CloneId` taking that as an parameter.